src/main/kotlin/com/github/spotbugs/snom/SpotBugsTask.kt

```diff
@@ -311,6 +311,8 @@ abstract class SpotBugsTask : DefaultTask(), VerificationTask {
                 "text" -> objects.newInstance(SpotBugsTextReport::class.java, name, objects, this)
                 "sarif" -> objects.newInstance(SpotBugsSarifReport::class.java, name, objects, this)
                 else -> throw InvalidUserDataException("$name is invalid as the report name")
+            }.also {
+                (outputs as org.gradle.api.tasks.TaskOutputs).file(it.outputLocation)
```

Comment:
This cast is necessary to ensure we depend on no Gradle internal API. If we have no cast here, ArchUnit will warn that this line depends on an internal API `InternalTaskPutputs.file(Object`).